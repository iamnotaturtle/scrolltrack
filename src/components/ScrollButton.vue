<template>
  <div class="scroll-button" @click="clicked()">
    <div class="percentage"/>
    <slot class="scroll-content" name="content"/>
  </div>
</template>

<script lang="ts">
import {Component, Vue} from 'vue-property-decorator';

@Component
export default class ScrollButton extends Vue {

  clicked(): void {
    this.$emit('clicked!');
  }
}
</script>

<style scoped lang="scss">
.scroll-button {
  background: #97c83e;
  width: 50px;
  height: 50px;
  border-radius: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  text-decoration: none;
  font-size: 35px;
  font-weight: bold;
  color: white;
}

.percentage {
	overflow: hidden;
	position: absolute;
	top: -1em; 
  right: 50%; 
  bottom: 50%; 
  left: -1em;
	transform-origin: 100% 100%;
	transform: rotate(45deg) skewX(30deg);
}

.percentage:before {
	box-sizing: border-box;
	display: block;
	border: solid 1em navy;
	width: 200%; 
  height: 200%;
	border-radius: 50%;
	transform: skewX(-30deg);
	content: '';
}
</style>
