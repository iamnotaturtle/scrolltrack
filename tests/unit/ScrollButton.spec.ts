import {shallowMount} from '@vue/test-utils';
import ScrollButton from '@/components/ScrollButton.vue';

describe('ScrollButton.vue', () => {
  it('emits correct scroll position', () => {
    expect(true).toBe(true);
  });
  it('renders content when passed in', () => {
    const wrapper = shallowMount(ScrollButton, {
      stubs: {
        ProgressBar: true,
      },
      slots: {
        content: '<div class="test">test</div>',
      },
    });
    expect(wrapper.contains('.test')).toBe(true);
  });
  it('renders correct progress bar color', () => {
    const wrapper = shallowMount(ScrollButton, {
      stubs: {
        ProgressBar: true,
      },
      propsData: {
        progressColor: 'purple',
      },
    });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
