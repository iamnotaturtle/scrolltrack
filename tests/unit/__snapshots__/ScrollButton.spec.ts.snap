// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollButton.vue renders correct progress bar color 1`] = `
<div
  class="scroll-button"
>
   
  <progressbar-stub
    color="purple"
  />
</div>
`;
